xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiAttachmentContent.java

```diff
@@ -318,6 +334,23 @@ public void setContent(InputStream is, int len) throws IOException
      * @since 2.6M1
      */
     public void setContent(InputStream is) throws IOException
+    {
+        setContent(is, false);
+    }
+
+    /**
+     * Set the content of the attachment from an InputStream.
+     *
+     * @param is the input stream that will be read
+     * @param skipDirty don't change the dirty flag: use {@code true} when using this method for cloning an already
+     * existing attachment content.
+     * @throws IOException when an error occurs during streaming operation
+     * @since 17.2.0RC1
+     * @since 16.10.5
+     * @since 16.4.7
+     */
+    @Unstable
+    public void setContent(InputStream is, boolean skipDirty) throws IOException
```

Comment:
This looks too hackish to be public. That's generally not how we deal with dirty flags and clones (the dirty flag is generally put back to what it's supposed to be after the clone).